src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Extensions.CommandLineUtils;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class StaticAssetsIntegrationTest : MSBuildIntegrationTestBase, IClassFixture<BuildServerTestFixture>, IClassFixture<PackageTestProjectsFixture>, IAsyncLifetime
+    {
+        public StaticAssetsIntegrationTest(
+            BuildServerTestFixture buildServer,
+            PackageTestProjectsFixture packageTestProjects,
+            ITestOutputHelper output)
+            : base(buildServer)
+        {
+            UseLocalPackageCache = true;
+            PackageTestProjects = packageTestProjects;
+            Output = output;
+        }
+
+        public PackageTestProjectsFixture PackageTestProjects { get; private set; }
+
+        public ITestOutputHelper Output { get; private set; }
+
+        [Fact]
+        [InitializeTestProject("AppWithPackageAndP2PReference")]
+        public async Task Build_GeneratesAspNetCoreStaticAssetsManifest_Success_CreatesManifest()
+        {
+            // For some reason when using a custom package cache the imports won't get added on
+            // the initial restore, so we restore the packages ourselves.
+            await DotnetMSBuild("Restore");
+
+            var expectedManifest = GetExpectedManifest();
+
+            var result = await DotnetMSBuild("Build");
+
+            Assert.BuildPassed(result);
+
+            // GenerateAspNetCoreStaticAssetsManifest should generate the manifest and the cache.
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.xml");
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.cache");
+
+            var path = Assert.FileExists(result, OutputPath, "AppWithPackageAndP2PReference.dll");
+            var assembly = Assert.ContainsEmbeddedResource(path, "Microsoft.AspNetCore.StaticAssets.xml");
+            using (var reader = new StreamReader(assembly))
+            {
+                var data = await reader.ReadToEndAsync();
+                Assert.Equal(expectedManifest, data);
```

Comment:
Can we deserialize a .NET object and compare that? We don't care that the file content needs to be identical, just that it needs to have the right content so anything that avoids exact matches makes it easier to maintain.

Reply:
I used to do that, but I was getting errors on OSX (due to a bug in the manifest extraction code) and that made it impossible to debug. This way is easier.

Reply:
changed